packages/augur-sdk/src/state/db/BaseSyncableDB.ts

```diff
@@ -0,0 +1,92 @@
+import { ParsedLog } from '@augurproject/types';
+import * as _ from 'lodash';
+import { Augur } from '../../Augur';
+import { BaseDocument } from './AbstractTable';
+import { DB } from './DB';
+import { RollbackTable } from './RollbackTable';
+
+export interface Document extends BaseDocument {
+  blockNumber: number;
+}
+
+/**
+ * Stores event logs for non-user-specific events.
+ */
+export class BaseSyncableDB extends RollbackTable {
+  protected eventName: string;
+
+  constructor(
+    augur: Augur,
+    protected db: DB,
+    networkId: number,
+    eventName: string,
+    dbName: string = eventName,
+  ) {
+    super(networkId, augur, dbName, db);
+    this.eventName = eventName;
+
+    db.notifySyncableDBAdded(this);
+    this.rollingBack = false;
+  }
+
+  async addNewBlock(blocknumber: number, logs: ParsedLog[]): Promise<number> {
+    // don't do anything until rollback is complete. We'll sync back to this block later
+    if (this.rollingBack) {
+      return -1;
```

Comment:
does this -1 make it to `getLogs`? What the the source of the hexify issue? I think it might have gotten fixed recently

Reply:
The case where that happens is just slightly different and was from a case where we'd double fetch the current block from the eth node, and sometimes fail. On failure it would return -1 and we'd roll forward with that as current block.

Also don't love the sentinel value here not sure if that can be changed.

Reply:
The value isn't used. Could just be an early return.